src/main/java/org/testng/TestRunner.java

```diff
@@ -254,6 +254,9 @@ private void init() {
 
     initListeners();
     addConfigurationListener(m_confListener);
+    for (IConfigurationListener cl : m_configuration.getConfigurationListeners()) {
```

Comment:
@missedone - Thanks for helping me understand. I will spend a bit more time on TestNG-remote to get a better understanding based on your explanation.

Apart from that query I didn't have any other observations.